src/btreemap/proptests.rs

```diff
@@ -62,3 +64,35 @@ proptest! {
         });
     }
 }
+
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq)]
+pub struct StorablePrincipal(pub Principal);
+
+impl Storable for StorablePrincipal {
+    fn to_bytes(&self) -> std::borrow::Cow<[u8]> {
+        self.0.as_ref().into()
+    }
+
+    fn from_bytes(bytes: std::borrow::Cow<[u8]>) -> Self {
+        Self(Principal::from_slice(bytes.as_ref()))
+    }
+    const BOUND: Bound = Bound::Bounded {
+        max_size: 29,
+        is_fixed_size: true,
```

Comment:
Principals aren't fixed-size
```suggestion
        is_fixed_size: false,
```